pkg/gather/service/analyze.go

```diff
@@ -0,0 +1,133 @@
+package service
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// AnalyzeGatherBundle will analyze the bootstrap gather bundle at the specified path.
+// Analysis will be logged.
+// Returns an error if there was a problem reading the bundle.
+func AnalyzeGatherBundle(bundlePath string) error {
+	// open the bundle file for reading
+	bundleFile, err := os.Open(bundlePath)
+	if err != nil {
+		return errors.Wrap(err, "could not open the gather bundle")
+	}
+	defer bundleFile.Close()
+	return analyzeGatherBundle(bundleFile)
+}
+
+func analyzeGatherBundle(bundleFile io.Reader) error {
+	// decompress the bundle
+	uncompressedStream, err := gzip.NewReader(bundleFile)
+	if err != nil {
+		return errors.Wrap(err, "could not decompress the gather bundle")
+	}
+	defer uncompressedStream.Close()
+
+	// read through the tar for relevant files
+	tarReader := tar.NewReader(uncompressedStream)
+	var releaseImageAnalysis *analysis
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return errors.Wrap(err, "encountered an error reading from the gather bundle")
+		}
+		if header.Typeflag != tar.TypeReg {
+			continue
+		}
+		filenameParts := strings.SplitN(header.Name, "/", 2)
```

Comment:
This seems more like a path than parts of a filename. 

Also a comment might help here about why we're splitting into 2. I think we're dropping the arbitrary log bundle filename.